pkg/kapp/cmd/app/apply_flags.go

```diff
@@ -55,6 +55,8 @@ func (s *ApplyFlags) SetWithDefaults(prefix string, defaults ApplyFlags, cmd *co
 
 	cmd.Flags().DurationVar(&s.WaitingChangesOpts.Timeout, prefix+"wait-timeout",
 		mustParseDuration("15m"), "Maximum amount of time to wait in wait phase")
+	cmd.Flags().DurationVar(&s.WaitingChangesOpts.ResourceTimeout, prefix+"wait-resource-timeout",
+		mustParseDuration("0"), "Maximum amount of time to wait for a resource in wait phase. 0s is the default value which indicates no timeout")
```

Comment:
`Maximum amount of time to wait for a resource in wait phase. 0s is the default value which indicates no timeout` -> `Maximum amount of time to wait for a resource in wait phase (0s means no timeout)`

Reply:
@cppforlife can be have default value as `0` instead of `0s` because time parser is treating differently if we give `0s` and the condition I have put in `pkg/kapp/clusterapply/waiting_changes.go` line 77 will work only for `0` not for `0s` if we are planning to have it as `0s` then we need to change the condition.

Reply:
how is it different? https://play.golang.org/p/5QqqYr9Iy6D